editor/animation_bezier_editor.cpp

```diff
@@ -1520,7 +1520,8 @@ void AnimationBezierTrackEdit::_zoom_callback(float p_zoom_factor, Vector2 p_ori
 		// Alternate zoom (doesn't affect timeline).
 		timeline_v_zoom = CLAMP(timeline_v_zoom * p_zoom_factor, 0.000001, 100000);
 	} else {
-		timeline->_zoom_callback(p_zoom_factor, p_origin, p_event);
+		float zoom_factor = p_zoom_factor > 1.0 ? AnimationTimelineEdit::SCROLL_ZOOM_FACTOR : 2.0 - AnimationTimelineEdit::SCROLL_ZOOM_FACTOR;
```

Comment:
If there is no need to calculate dynamically here, you should make them two constants like `SCROLL_ZOOM_FACTOR_H`, `SCROLL_ZOOM_FACTOR_V`.

Reply:
The problem is, that `p_zoom_factor` is either greater than 1 when zooming in (I think) and less than 1 when zooming out. The amount by which it is smaller or greater than 1 is defined by the `scroll_zoom_factor` set for the `Range`, e.g. if it is `1.1`, then the value will either be `0.9` or `1.1`. For the vertical zoom, this value used to be just the default, for the horizontal we have the constant `AnimationTimelineEdit::SCROLL_ZOOM_FACTOR`. Since we cannot set the zoom factor directly for the Bezier editor, since this would affect the timeline and the vertical zoom equally, but we have to calculate what the zoom would be, if it were using the constant, i.e. `SCROLL_ZOOM_FACTOR` (`1.02`) or  `2.0 - SCROLL_ZOOM_FACTOR` (0.98).

Reply:
I mean you should write as the following:

```cpp
float AnimationTimelineEdit::SCROLL_ZOOM_FACTOR_A = 1.02f;
float AnimationTimelineEdit::SCROLL_ZOOM_FACTOR_B = 0.98f;

float zoom_factor = p_zoom_factor > 1.0 ? AnimationTimelineEdit::SCROLL_ZOOM_FACTOR_A : AnimationTimelineEdit::SCROLL_ZOOM_FACTOR_B;
```

Reply:
ok, I added it as an additional constant.